spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/DataTables.groovy

```diff
@@ -446,100 +252,235 @@ local ; 1
+        1 $sepA _
+        2 $sepA _
+        combined:
+        b $sepB _
+        3 $sepB _
+        combined:
+        c $sepA _
+        4 $sepA _
+        5 $sepA _
+        6 $sepA _
+      }
+    """
 
     then:
-    thrown Exception
+    results.testsStartedCount == 1 + 6
+    results.testEvents().started().list().testDescriptor.displayName == [
+      'a feature (#a #b #c)',
+      'a feature (1 3 4)',
+      'a feature (2 3 4)',
+      'a feature (1 3 5)',
+      'a feature (2 3 5)',
+      'a feature (1 3 6)',
+      'a feature (2 3 6)'
+    ]
+
+    where:
+    sepA | sepB | separators
+    '|'  | '|'  | 'pipes'
+    ';'  | ';'  | 'semicolons'
+    '|'  | ';'  | 'mixed separators'
+  }
+
+  def 'combined data table columns can use previous data table column values within one table'() {
+    expect:
+    a == b
+
+    where:
+    i << (1..6)
+
+    a | b
+    1 | a
+    2 | a
+    combined:
+    c << (1..3)
   }
 
-  def "cells can't reference themselves"() {
+  @PendingFeature(reason = 'previous column access across tables does not yet work as expected with cross-multiplication')
+  def 'combined data table columns can use previous data table column values across tables'() {
     when:
-    runner.runFeatureBody '''
+    def results = runner.runFeatureBody '''
       expect:
-      false
+      true
+      a == b
+      c == "x$i"
 
       where:
-      a | b
-      1 | b + 1
+      x    | _
+      'x1' | _
+      'x2' | _
+      'x3' | _
+      'x4' | _
+      'x5' | _
+      'x6' | _
+
+      i << (1..6)
+
+      a | _
+      1 | _
+      2 | _
+      combined:
+      b | _
+      a | _
+      combined:
+      c | _
+      x | _
+      x | _
+      x | _
     '''
 
     then:
-    thrown Exception
+    results.testsSucceededCount == 1 + 6
   }
 
-  def 'data tables can be referenced from following variables'() {
-    expect:
-    c == 3
+  def 'data tables with different widths can be combined'() {
+    when:
+    def results = runner.runSpecBody '''
+      def 'a feature (#a #b #c #d #e)'() {
+        expect:
+        true
 
-    where:
-    a | b
-    1 | 2
-    1 | a + 1
+        where:
+        a | b
+        1 | 'b'
+        2 | 'b'
+        combined:
+        c | d   | e
+        3 | 'd' | 'e'
+      }
+    '''
 
-    c = b + 1
+    then:
+    results.testsStartedCount == 1 + 2
+    results.testEvents().started().list().testDescriptor.displayName == [
+      'a feature (#a #b #c #d #e)',
+      'a feature (1 b 3 d e)',
+      'a feature (2 b 3 d e)'
+    ]
   }
 
-  def "rows must have same number of elements as header"() {
+  def 'data tables starting with table separator can be combined'() {
     when:
-    compiler.compileFeatureBody """
-expect:
-true
+    def results = runner.runSpecBody '''
+      def 'a feature (#a #b #c #d #e)'() {
+        expect:
+        true
 
-where:
-a | b | c
-1 | 2 | 3
-4 | 5
-    """
+        where:
+        a | b
+        1 | 'b'
+        2 | 'b'
+        combined:
+        _____________
```

Comment:
The `combined:` block is always stuck right at the previous block. I think this is not required, but we have no test that shows otherwise, i.e., with a blank line. Also, test if the table separator works above as well.

Reply:
> The combined: block is always stuck right at the previous block. I think this is not required

Correct

> but we have no test that shows otherwise, i.e., with a blank line

I changed one of the `DataSpec` tests to have blank lines, then it is tested and also shown in the docs.

> Also, test if the table separator works above as well.

Added and make it work. :-D